proposed/tracing.md

```diff
@@ -10,14 +10,103 @@ interfaces and functionality first.
+
+To track the time a piece of work takes, a user will call `SpanInterface::activate()`.
+This will set the start time and push the current span into whichever context propagation system the provider chooses.
+Users MAY call `SpanInterface::start()` to populate the span with a start time, but not pushed into context propagation.
+Once complete, the user would call SpanInterface::finish() to set the end-time and “pop” the span from the current
+stack.
+
+Spans MUST support creating a child span via `SpanInterface::createChild(string $spanName): SpanInterface`.
+This SHOULD use `TracerInterface::createSpan()`, then set the subsequent span's parent appropriately.
+
+A typical use would look like the following:
+
+```php
+function imgResize($size=100) {
+    $span = $this->tracer->createSpan('image.resize')
+        ->setAttribute('size',$size)
+        ->activate();
+
+    try{
+      //Resize the image
+      return $resizedImage;
+    
+    } catch (Exception $e) {
+        // Ideally, you would attach the exception to the span here
+        $span->setStatus(SpanInterface::STATUS_ERROR)
+             ->addException($e);
+    } finally {
+        $span->finish();
+    }    
+}
+```
+
+This PSR does not dictate how a span’s internals should be represented, other than it MUST implement the SpanInterface.
+However, it is RECOMMENDED to conform to the [W3C TraceContext specification](https://www.w3.org/TR/trace-context/).
+Providers MUST implement the `SpanInterface::toTraceContextHeaders()` method.
+At a minimum, this should return an empty array.
+Most commonly, providers will return traceParent and baggage headers to pass on to child services.
+
+Providers MUST add the following function signatures to allow data retrieval from their span:
+- `getAttribute(string $key): null|string|int|float|bool|Stringable;`
+- `getAttributes(): iterable;`
+- `getParent(): SpanInterface;`
+- `getChildren(): array;`
+
+The purpose of these functions is to allow data to be read from providers’ spans in a clear, uniform manner.
+`SpanInterface::getParent` MAY return null if no parent is present or the span has been instantiated outside the relevant adapter’s methods.
+However, a reasonable attempt SHOULD be made to return a `SpanInterface` where possible.
+Similarly, `SpanInterface::getChildren()` MAY return an empty array if child spans have been created outside the relevant adapter's methods.
+For continual use of a secondary provider, it is RECOMMENDED to create a separate Tracer and use the `MultiTracer` described rather than relying on these methods.
+
+## 2.2 Tracer
+
+See: [AAllport/psr-tracing - TracerInterface.php](https://github.com/AAllport/psr-tracing/blob/main/src/TracerInterface.php)
+
+SDKs are expected to provide a concrete Tracer via whichever methods are most appropriate for the frameworks they support (eg, PSR-11 container).
+Libraries using this PSR SHOULD implement [TracerAwareTrait](#31-tracerawaretrait).
```

Comment:
I guess here we should mention `TracerAwareInterface` instead of the trait.
I believe that it might be better to avoid the AwareInterface pattern and have just a `TracerManager` class that can be global. My assumption is that providers are provided as a php extension or as a wrapper over that so exposing it as an global class might fit better. Or inject it in the container and obtain it where is needed.

Reply:
I would personally avoid a `TracerManager`: DI containers can handle it well enough.

Maybe this can be rephrased as:
`It is RECOMMENDED for the libraries using this PSR implement TracerAwareInterface in order to ease the injection of a tracer` 

or something similar